library/std/src/process.rs

```diff
@@ -2158,26 +2148,30 @@ impl Termination for ! {
 }
 
 #[stable(feature = "termination_trait_lib", since = "1.61.0")]
-impl<E: fmt::Debug> Termination for Result<!, E> {
+impl Termination for Infallible {
     fn report(self) -> ExitCode {
-        let Err(err) = self;
-        eprintln!("Error: {err:?}");
-        ExitCode::FAILURE
+        // Doesn't matter, this is unreachable
+        ExitCode::SUCCESS
     }
 }
 
 #[stable(feature = "termination_trait_lib", since = "1.61.0")]
-impl<E: fmt::Debug> Termination for Result<Infallible, E> {
+impl Termination for ExitCode {
+    #[inline]
     fn report(self) -> ExitCode {
-        let Err(err) = self;
-        Err::<!, _>(err).report()
+        self
     }
 }
 
 #[stable(feature = "termination_trait_lib", since = "1.61.0")]
-impl Termination for ExitCode {
-    #[inline]
+impl<T: Termination, E: fmt::Debug> Termination for Result<T, E> {
```

Comment:
similarly this impl is technically new but is backfilling removed impls that were since 1.61.0 so..?

Reply:
Imo backdating them both seems correct since they're both representing existing impls from that release even if they both add new impls as well.

Reply:
It's for the `stable_features` diagnostic.

```console
warning: the feature `pin` has been stable since 1.33.0 and no longer requires an attribute to enable
 --> src/main.rs:1:12
  |
1 | #![feature(pin)]
  |            ^^^
  |
  = note: `#[warn(stable_features)]` on by default
```

which means it is pretty much meaningless for impls, since you don't get impls by enabling a feature.